build.gradle

```diff
@@ -225,20 +225,22 @@ artifacts {
     archives makeMdk
 }
 
-task ciWriteBuildNumber << {
-    def file = file("src/main/java/net/minecraftforge/common/ForgeVersion.java");
-    def bn = System.getenv("BUILD_NUMBER")?:project.ext.properties.buildNumber?:0;
-    def outfile = "";
-    def ln = "\n"; //Linux line endings because we're on git!
-
-    file.eachLine{ String s ->
-        if (s.matches("^    public static final int buildVersion    = [\\d]+;\$"))
-            s = "    public static final int buildVersion    = ${bn};";
-        if (s.matches('^    public static final String mcVersion = "[^\\"]+";'))
-            s = "    public static final String mcVersion = \"${minecraft.version}\";";
-        outfile += (s+ln);
+task ciWriteBuildNumber {
+    doLast {
```

Comment:
Please explain the change from `<<` to `doLast`

Reply:
to solve gradle warnings

Reply:
It solves deprecation warnings on Gradle 3.2 and Up, should Forge ever officially change to 3.2+